pkg/network/infraconfigurators/bridge.go

```diff
@@ -197,6 +193,17 @@ func (b *BridgePodNetworkConfigurator) createBridge() error {
 		return err
 	}
 
+	//replace podNic's mac with bridge's mac before linking podIface as master on bridge
+	brMac, err := b.handler.GetMac(b.bridgeInterfaceName)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to access bridge")
+		return err
+	}
+
+	if err := b.handler.SetMac(b.podNicLink.Attrs().Name, brMac.String()); err != nil {
+		return err
+	}
```

Comment:
This is equivalent to directly doing:
```
link, err := h.LinkByName(ifaceName)
if err != nil {
	return nil, err
}
return netlink.LinkSetHardwareAddr(link, link.Attrs().HardwareAddr)
```

So why do we need those helper functions? They seem to complicate things.

There is probably an issue with using `nelink` directly, but that is true for both versions.